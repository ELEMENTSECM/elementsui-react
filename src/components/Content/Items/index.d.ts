import * as React from "react";

export interface ItemsProps {
	/**
	 * CSS class name for container element holding items
	 */
	containerClassName?: string;
	/**
	 * CSS class name of item element
	 */
	itemClassName?: string;
	/**
	 * Function to return contents of item element
	 */
	renderItem: (...args: any[]) => any;
	/**
	 * Array of items
	 */
	items: any[];
	/**
	 * Function called when remove button is clicked on item element
	 */
	onItemRemove: (...args: any[]) => any;
}

declare const Items: React.SFC<ItemsProps>;

export default Items;
