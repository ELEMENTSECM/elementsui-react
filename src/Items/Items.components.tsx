import styled from "styled-components";
import { Well } from "react-bootstrap";

export const ItemsContainer = styled(Well)`
	padding: 2px;
	min-height: 26px;
	background-color: #f8f8f8;

	> ul.eui-items__list {
		max-height: 105px;
		overflow-y: auto;
		display: block;
		padding: 0;
		box-sizing: border-box;
		list-style: none;
		margin: 0;
		width: 100%;
		text-overflow: ellipsis;
		white-space: nowrap;

		> li.eui-items__list__item {
			margin: 0 0 1px 3px;
			padding: 0 6px 0 3px;
			float: left;
			border-radius: 1px;
			border: 1px solid #cccaca;
			outline: solid 1px transparent;
			background-color: #f1eeee;
			text-shadow: 0 0 3px #fffefe;
			color: #333;
			cursor: default;

			&:hover {
				background-color: #fbfbfb;
				border: 1px solid #c3c2c2;
				cursor: pointer;
			}

			> button.eui-items__list__item__remove {
				color: #676767;
				padding: 0 1px;
				margin-right: 3px;
				cursor: pointer;
				display: inline-block;
				font-weight: bold;
				background: none;
				border: none;

				&:hover {
					transform: scale(1.2);
					color: #560404;
				}

				&:focus {
					outline: none;
				}
			}
		}
	}
`;
