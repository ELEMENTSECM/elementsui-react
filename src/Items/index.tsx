export { default, ItemsProps } from "./Items";
