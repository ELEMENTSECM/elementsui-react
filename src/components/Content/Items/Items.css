.eui-items {
    padding: 2px;
    min-height: 26px;
    background-color: #f8f8f8;
}

.eui-items__list {
    max-height: 105px;
    overflow-y: auto;
    display: block;
    padding: 0;
    box-sizing: border-box;
    list-style: none;
    margin: 0;
    width: 100%;
    text-overflow: ellipsis;
    white-space: nowrap;
}

.eui-items__list__item {
    margin: 0 0 1px 3px;
    padding: 0 6px 0 3px;
    float: left;
    border-radius: 1px;
    border: 1px solid #cccaca;
    outline: solid 1px transparent;
    background-color: #f1eeee;
    text-shadow: 0 0 3px #fffefe;
    color: #333;
    cursor: default;
}

.eui-items__list__item:hover {
    background-color: #fbfbfb;
    border: 1px solid #c3c2c2;
    cursor: pointer;
}

.eui-items__list__item__remove {
    color: #676767;
    padding: 0 1px;
    margin-right: 3px;
    cursor: pointer;
    display: inline-block;
    font-weight: bold;
    background: none;
    border: none;
}

.eui-items__list__item__remove:hover {
    transform: scale(1.2);
    color: #560404;
}

.eui-items__list__item__remove:focus {
    outline: none;
}