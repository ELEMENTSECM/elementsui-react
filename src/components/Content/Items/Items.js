import * as React from "react";
import PropTypes from "prop-types";
import { Well } from "react-bootstrap";
import classNames from "classnames";

export default function Items(props) {
	const { items, renderItem, onItemRemove, containerClassName, itemClassName } = props;
	return (
		<Well bsSize="sm" className={classNames("eui-items", containerClassName)}>
			<ul className="eui-items__list">
				{items.map((item, index) => (
					<li key={index} className={classNames("eui-items__list__item", itemClassName)}>
						<button
							type="button"
							className="eui-items__list__item__remove"
							onClick={() => onItemRemove(item)}
						>
							Ã—
						</button>
						<span>{renderItem(item)}</span>
					</li>
				))}
			</ul>
		</Well>
	);
}

Items.propTypes = {
	/** CSS class name for container element holding items  */
	containerClassName: PropTypes.string,
	/** CSS class name of item element */
	itemClassName: PropTypes.string,
	/** Function to return contents of item element */
	renderItem: PropTypes.func.isRequired,
	/** Array of items */
	items: PropTypes.array.isRequired,
	/** Function called when remove button is clicked on item element */
	onItemRemove: PropTypes.func.isRequired
};
