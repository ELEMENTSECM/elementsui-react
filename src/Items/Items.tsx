import * as React from "react";
import { Well } from "react-bootstrap";
import classNames from "classnames";
import styled from "styled-components";
import { ItemsContainer } from "./Items.components";

export interface ItemsProps<TItem> {
	/** CSS class name for container element holding items  */
	containerClassName?: string;
	/** CSS class name of item element */
	itemClassName?: string;
	/** Function to return contents of item element */
	renderItem: (item: TItem) => React.ReactNode;
	/** Array of items */
	items: TItem[];
	/** Function called when remove button is clicked on item element */
	onItemRemove: (item: TItem) => any;
}

export default function Items<T>(props: ItemsProps<T>) {
	const { items, renderItem, onItemRemove, containerClassName, itemClassName } = props;
	return (
		<ItemsContainer bsSize="sm" className={classNames("eui-items", containerClassName)}>
			<ul className="eui-items__list">
				{items.map((item, index) => (
					<li key={index} className={classNames("eui-items__list__item", itemClassName)}>
						<button
							type="button"
							className="eui-items__list__item__remove"
							onClick={() => onItemRemove(item)}
						>
							Ã—
						</button>
						<span>{renderItem(item)}</span>
					</li>
				))}
			</ul>
		</ItemsContainer>
	);
}
