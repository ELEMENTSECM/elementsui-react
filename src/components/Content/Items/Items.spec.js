import * as React from "react";
import { shallow } from "enzyme";
import Items from "./index";
import _ from "lodash";

describe("Items", () => {
	test("displays correctly", () => {
		const wrapper = shallow(<Items items={["test item"]} renderItem={x => x.toString()} onItemRemove={_.noop} />);
		expect(wrapper).toMatchSnapshot();
	});
});
