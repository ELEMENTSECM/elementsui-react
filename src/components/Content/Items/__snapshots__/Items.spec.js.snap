// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items displays correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Items
    items={
      Array [
        "test item",
      ]
    }
    onItemRemove={[Function]}
    renderItem={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bsClass": "well",
      "bsSize": "sm",
      "children": <ul
        className="eui-items__list"
      >
        <li
          className="eui-items__list__item"
        >
          <button
            className="eui-items__list__item__remove"
            onClick={[Function]}
            type="button"
          >
            ×
          </button>
          <span>
            test item
          </span>
        </li>
      </ul>,
      "className": "eui-items",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="eui-items__list__item"
          >
            <button
              className="eui-items__list__item__remove"
              onClick={[Function]}
              type="button"
            >
              ×
            </button>
            <span>
              test item
            </span>
          </li>,
        ],
        "className": "eui-items__list",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="eui-items__list__item__remove"
                onClick={[Function]}
                type="button"
              >
                ×
              </button>,
              <span>
                test item
              </span>,
            ],
            "className": "eui-items__list__item",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "×",
                "className": "eui-items__list__item__remove",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "×",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "test item",
              },
              "ref": null,
              "rendered": "test item",
              "type": "span",
            },
          ],
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "well",
        "bsSize": "sm",
        "children": <ul
          className="eui-items__list"
        >
          <li
            className="eui-items__list__item"
          >
            <button
              className="eui-items__list__item__remove"
              onClick={[Function]}
              type="button"
            >
              ×
            </button>
            <span>
              test item
            </span>
          </li>
        </ul>,
        "className": "eui-items",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="eui-items__list__item"
            >
              <button
                className="eui-items__list__item__remove"
                onClick={[Function]}
                type="button"
              >
                ×
              </button>
              <span>
                test item
              </span>
            </li>,
          ],
          "className": "eui-items__list",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="eui-items__list__item__remove"
                  onClick={[Function]}
                  type="button"
                >
                  ×
                </button>,
                <span>
                  test item
                </span>,
              ],
              "className": "eui-items__list__item",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "×",
                  "className": "eui-items__list__item__remove",
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "×",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "test item",
                },
                "ref": null,
                "rendered": "test item",
                "type": "span",
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
